### PR TITLE
Implements S3 proxy

### PR DESCRIPTION
**Pull request by ashfurrow**
### Description

This PR adds a small express server to the `heroku-redirect-to-s3` repo. We'll plan on keeping this running more-or-less indefinitely while we wait for all our users to update to newer versions of the iOS app (/cc @joeyAghion). I'll need to update the README here and on `master` too.

Here's it working locally with Eigen's `6.6.4-2019.05.24.09-submission` tag, before we adopted the new S3 Echo, on a fresh install:

Screen Shot 2020-12-04 at 12 36 14

You can see logs where the simulator has made HEAD and GET requests:

Screen Shot 2020-12-04 at 12 35 52

And against a fresh Heroku app I've set up:

Screen Shot 2020-12-04 at 12 50 30
Screen Shot 2020-12-04 at 12 51 14

## Test Plan

Once this is merged, I'll try to deploy this to our Echo Heroku app. TestFlight will let me install an older build of the app on my device so we can test that everything works. If it doesn't, we can roll back in Heroku.

### PR Checklist (tick all before merging)

- [x] I have not stored any sensitive keys/values/information in `Echo.json5` and in CI, or I have not changed anything in that file.

### CODE REVIEW

**Review by pvinis — APPROVED** [review]
Totally makes sense! Nice!

File: package.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "echo",
+  "version": "0.0.1",
+  "description": "A server to redirect Heroku requests to the new S3 bucket.",
+  "main": "index.js",
+  "repository": "https://github.com/artsy/echo.git",
```
[the remark this answers is not in the document]
Comment (ashfurrow) [reply package.json:6]: Oooooo yes good idea!

**Review by joeyAghion — APPROVED** [review]
Redirect is appealing because we won't have an old version of this application deployed out there, and 2 different versions of the configuration data it serves.

About that, if I'm reading the code correctly, this is more _proxying_ to S3 than redirecting, right?

One slight behavior change here is that the Heroku app will stop requiring authorization. I think that's fine, since that's similar to the S3 origin, but wanted to confirm it was intentional. This may just manifest as 404s instead of 401s responding to random internet traffic.

**Comment by ashfurrow** [comment]
Yes exactly – even if we need to have _something_ at `https://echo-api-production.herokuapp.com/` I'd prefer it to have it deployable and serving the latest config. That way older apps will still respect Echo changes, for example, like the `KillSwitchBuildMinimum`.

> About that, if I'm reading the code correctly, this is more proxying to S3 than redirecting, right?

Ah, yes good point. I'll make sure to use the correct term when writing the docs.

The lack of authorization was intentional, following the pattern we established with the S3 bucket that we built without auth. We could add a simple "is the header specified at all?" check, but I'm okay with the 401's/404's.

**Comment by ashfurrow** [comment]
We successfully deployed this – I'll update the readme in a subsequent PR.

**Pull request merged**